### ISSUE TITLE
Is go-pry supposed to run standalone

### ISSUE THREAD

**Issue by tompave** [d4l3k/go-pry#49]
Apologies if this is already explained elsewhere, but it's not clear from the examples in the readme if it's possible to run `go-pry` as a standalone REPL.

When I try I get this error, but I'm not sure if it's the normal behaviour or a problem with my setup.

```
$ go get github.com/d4l3k/go-pry
$ go install github.com/d4l3k/go-pry
$ which go-pry
....../go/bin/go-pry
$ go-pry -i="fmt,math,strconv"
panic: failed to TypeCheck: /var/folders/mw/hh951k897lv1c8nchgxyy9zw0000gp/T/pry101925011/main.go:3:2: could not import github.com/d4l3k/go-pry/pry (can't find import: "github.com/d4l3k/go-pry/pry")

goroutine 1 [running]:
github.com/d4l3k/go-pry/pry.Apply(0xc4200ba200)
    /Users/Tom/source/go/src/github.com/d4l3k/go-pry/pry/pry.go:86 +0x228
main.main()
    /var/folders/mw/hh951k897lv1c8nchgxyy9zw0000gp/T/pry101925011/main.go:9 +0x3217
exit status 2
```

**Comment by tompave** [comment]
The same happens when running it inside a project with proper import and `pry.Pry()` call in the `main`.

**Comment by d4l3k** [comment]
That's an interesting error. Is $GOPATH set?

**Comment by d4l3k** [comment]
/ can you paste the output from `go env` ?

**Comment by tompave** [comment]
Hi, thank you for the quick answer. Sure.

Here it is.

```
$ go env
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/Tom/Library/Caches/go-build"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/Tom/source/go"
GORACE=""
GOROOT="/usr/local/go"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/mw/hh951k897lv1c8nchgxyy9zw0000gp/T/go-build188710889=/tmp/go-build -gno-record-gcc-switches -fno-common"
```
I can also confirm that `go-pry` was available in `$GOPATH/bin/go-pry`.

Also:

```
$ go version
go version go1.10.1 darwin/amd64
```

I installed go using the macOS installer from golang.org, and I'm running it on macOS 10.13.4.

**Comment by d4l3k** [comment]
Might try and run `go get -u github.com/d4l3k/go-pry`.

I'm having a hard time reproducing it on Linux and don't have a mac.

**Comment by tompave** [comment]
> Might try and run `go get -u github.com/d4l3k/go-pry`.

I've tried that. It got correctly recompiled but the problem persists:

```
$ go-pry -i="fmt,math,strconv"
panic: failed to TypeCheck: /var/folders/mw/hh951k897lv1c8nchgxyy9zw0000gp/T/pry768038902/main.go:3:2: could not import github.com/d4l3k/go-pry/pry (can't find import: "github.com/d4l3k/go-pry/pry")

goroutine 1 [running]:
github.com/d4l3k/go-pry/pry.Apply(0xc4200ce200)
    /Users/Tom/source/go/src/github.com/d4l3k/go-pry/pry/pry.go:86 +0x228
main.main()
    /var/folders/mw/hh951k897lv1c8nchgxyy9zw0000gp/T/pry768038902/main.go:10 +0x3f4f
exit status 2
```

> I'm having a hard time reproducing it on Linux and don't have a mac.

Do you know if any other mac user has the same problem? i.e. `go-pry` doesn't work on macOS?
I'm asking because since I opened the issue I changed my go installation: I moved from the [`golang.org` installer](https://golang.org/dl/) to a [package manager](https://brew.sh/) (for reasons unrelated to this issue). Since the problem is still there, I wonder if it's just a macOS thing.

> We could setup a time for me to remote in

Thanks for the offer :-), but I don't think it's necessary.

**Comment by Asafb26** [comment]
happens to me too, also on mac.
did you manage to make pry.Pry() works? Its looks like it ignore this line at all

**Comment by tompave** [comment]
No, I couldn't get it to work.

**Comment by d4l3k** [comment]
If you try putting a simple go file in one of the user temporary directories /var/folders/mw/*/T/* does it work?

```go
package main

import "github.com/d4l3k/go-pry/pry"

func main() {
pry.Pry()
}
```

Run with `go run main.go`

Might also try moving your `go` directory to be in your home folder. Ex `/Users/Tom/go`.

If one of those works it'll help me figure out what the cause might be. I suspect there's something weird going on with passing the environment variables on macOS. If just `~/go` works that's likely it.

**Comment by mitch-seymour** [comment]
I had the same issue running `go-pry` on macOS. Adding the `-i` flag to the `go install` command resolved it for me. cc/ @tompave @d4l3k

**Comment by tompave** [comment]
Thanks @mitch-seymour. Installing with `-i` fixed it for me too.

**Comment by kalashnikovisme** [comment]
Installing with `-i` didn't fix my problem :(

**Comment by barrio** [comment]
I confirm recompiling and installing doesn't fix it on ubuntu bionic with go 1.10.4.
